### PR TITLE
Support CSS embedded in HTML files

### LINKED ISSUE #3: Support CSS embedded in HTML files
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/atom-autoprefixer/issues/3)

It's currently supported by selecting the CSS manually. But I think we can do one better and do it automatically.

It might be possible to get all the CSS scopes and convert them.

https://discuss.atom.io/t/find-specific-scopes-in-a-file/12428
https://atom.io/docs/v0.209.0/behind-atom-scoped-settings-scopes-and-scope-descriptors

Could also maybe use: https://github.com/RebelMail/html-postcss

> There is a $40.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/atom-autoprefixer/issues/3).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/atom-autoprefixer/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

### PR DESCRIPTION
**Pull request by sasikanth513**
This should close #3

## Issue

Right now, if you want to autoprefix css inside html file, you need to select the css part of the file and then select `autoprefixer` command.

## Fix

As mentioned in [this comment](https://github.com/sindresorhus/atom-autoprefixer/issues/3#issue-28458112), using [RebelMail/html-postcss](https://github.com/RebelMail/html-postcss) package, we are able to detect css scope inside html and pass it to postcss.

Note: [RebelMail/html-postcss](https://github.com/RebelMail/html-postcss) package needs complete HTML to work, so if user selected part of css inside html file we're rolling back to using old process.

### CODE REVIEW

**Comment by sasikanth513** [comment]
I didn't know I can test linting issues in local till now for this package. So more commits than expected. 😄

File: index.js [hunk 1]
```diff
@@ -4,42 +4,55 @@ import postcss from 'postcss';
 import autoprefixer from 'autoprefixer';
 import postcssSafeParser from 'postcss-safe-parser';
 import postcssScss from 'postcss-scss';
+import HTMLPostCSS from 'html-postcss';
```
Comment (sindresorhus) [inline index.js]:
```suggestion
import HtmlPostCss from 'html-postcss';
```

File: index.js [hunk 2]
```diff
@@ -4,42 +4,55 @@ import postcss from 'postcss';
...
 
 	try {
-		const result = await postcss(autoprefixer(atom.config.get('autoprefixer'))).process(text, options);
-
-		result.warnings().forEach(x => {
-			console.warn(x.toString());
-			atom.notifications.addWarning('Autoprefixer', {
-				detail: x.toString()
+		let outCss;
+		// Package htmlPostCss requires complete html to process, selectedText doesn't work
```
Comment (sindresorhus) [inline index.js]:
```suggestion
		// The `html-postcss` package requires complete HTML to work, so `selectedText` doesn't work
```

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by sasikanth513** [comment]
sure, will do it.

**Comment by sasikanth513** [comment]
@sindresorhus fixed merge conflicts. Everything should work fine now.

**Comment by sindresorhus** [comment]
Prefixing the CSS in the HTML file works, but the command to remove prefixes does not work in the HTML file.

I just get an error message:

> Unexpected '!'. Escaping special characters with \ may help.

**Comment by sasikanth513** [comment]
@sindresorhus added option to remove prefixes from html file.

There seems to be some formatting issues here.

This is html file i'm testing on

Screenshot 2019-05-30 at 10 47 26 PM

it becomes this when autoprefix is done

Screenshot 2019-05-30 at 10 47 35 PM

and when we do `autoprefix:remove prefixes` this is how it becomes

Screenshot 2019-05-30 at 10 47 44 PM

**Comment by sasikanth513** [comment]
@sindresorhus seems like above issue is there even with the published version, it is not introduced by this PR.

Should I open new issue for this?

**Comment by sindresorhus** [comment]
While the issue is not introduced in this PR, I think it should still be fixed here as the feature is not really usable without.

**Comment by sasikanth513** [comment]
@sindresorhus sure, let me try this. Any inputs on where it might be the issue?

**Comment by sasikanth513** [comment]
setting `cascade` property as false removes that multiple tabs issue `(filter property in above screenshot)` but the issue with `:placeholder` property remains.

**Comment by sasikanth513** [comment]
just an update, I think formatting issues is because of `cheerio` package used by `html-postcss` package. 

I tried to use different approach using `first-man` npm package but I was having issues adding it to the project.

**Comment by sasikanth513** [comment]
Closing this as I can't figure out how to make indentation work.

**Pull request closed without merging**